qcore/geo.py

```diff
@@ -149,6 +149,57 @@ def ll2gp_multi(
     return xy
 
 
+def oriented_bearing_wrt_normal(
+    u: np.ndarray, v: np.ndarray, normal: np.ndarray
+) -> float:
+    """Compute the oriented bearing from u to v with respect to a normal.
+
+    This function is useful to calculate, for example, strike and dip
+    directions. The orientation is established via the right hand rule
+    (or refer to the diagram below).
+
+           v
+           ^
+           │
+           │
+           │
+           │
+           │
+           │<┐
+           │ └─┐
+           ╳─────────────>u
+          ╱
+         ╱
+        ╱
+       ╱
+    normal
+
+    Parameters
+    ----------
+    u : np.ndarray
+    v : np.ndarray
+    normal : np.ndarray
```

Comment:
A description for these could be great, not sure if there is a better variable name that could be used instead of u and v to better describe what your actually dealing with

Reply:
Best I can think of is changing u and v to from_direction and to_direction.